proxy/src/main/java/net/md_5/bungee/entitymap/EntityMap_1_8.java

```diff
@@ -56,95 +56,54 @@ public void rewriteClientbound(ByteBuf packet, int oldId, int newId)
         //Special cases
         int readerIndex = packet.readerIndex();
         int packetId = DefinedPacket.readVarInt( packet );
-        int packetIdLength = packet.readerIndex() - readerIndex;
-        if ( packetId == 0x0D /* Collect Item */ )
+        int readerIndexAfterPID = packet.readerIndex();
+        switch ( packetId )
         {
-            DefinedPacket.readVarInt( packet );
-            rewriteVarInt( packet, oldId, newId, packet.readerIndex() );
-        } else if ( packetId == 0x1B /* Attach Entity */ )
-        {
-            rewriteInt( packet, oldId, newId, readerIndex + packetIdLength + 4 );
-        } else if ( packetId == 0x13 /* Destroy Entities */ )
-        {
-            int count = DefinedPacket.readVarInt( packet );
-            int[] ids = new int[ count ];
-            for ( int i = 0; i < count; i++ )
-            {
-                ids[i] = DefinedPacket.readVarInt( packet );
-            }
-            packet.readerIndex( readerIndex + packetIdLength );
-            packet.writerIndex( readerIndex + packetIdLength );
-            DefinedPacket.writeVarInt( count, packet );
-            for ( int id : ids )
-            {
-                if ( id == oldId )
-                {
-                    id = newId;
-                } else if ( id == newId )
-                {
-                    id = oldId;
-                }
-                DefinedPacket.writeVarInt( id, packet );
-            }
-        } else if ( packetId == 0x0E /* Spawn Object */ )
-        {
-
-            DefinedPacket.readVarInt( packet );
-            int type = packet.readUnsignedByte();
-
-            if ( type == 60 || type == 90 )
-            {
-                packet.skipBytes( 14 );
-                int position = packet.readerIndex();
-                int readId = packet.readInt();
-                int changedId = readId;
+            case 0x1B /* Attach Entity */:
+                rewriteInt( packet, oldId, newId, readerIndexAfterPID + 4 );
+                break;
+            case 0x0D /* Collect Item */:
+                DefinedPacket.skipVarInt( packet );
+                rewriteVarInt( packet, oldId, newId, packet.readerIndex() );
+                break;
+            case 0x13 /* Destroy Entities */:
+                rewriteEntityIdArray( packet, oldId, newId, readerIndexAfterPID );
+                break;
+            case 0x0E /* Spawn Object */:
+                DefinedPacket.skipVarInt( packet );
+                int type = packet.readUnsignedByte();
 
-                if ( readId == oldId )
-                {
-                    packet.setInt( position, changedId = newId );
-                } else if ( readId == newId )
+                if ( type == 60 || type == 90 )
```

Comment:
Should probably just bring this method out for consistency

Reply:
When I did this I mostly followed intellij's duplicate code detection, thats why did not got its own method yet (because 1.9+ is different).
Will give this its own method tho.

Reply:
With a little bit of delay, done.